### PR TITLE
React to hosting

### PR DESCRIPTION
**Pull request by HaoK**
https://github.com/aspnet/Hosting/pull/210/files

Does music store actually use the ApplicationName for anything?  There's no easy way to set app name now.

cc @davidfowl @Praburaj

### CODE REVIEW

**Comment by Praburaj** [comment]
>Does music store actually use the ApplicationName for anything? There's no easy way to set app name now.

Not really.

File: src/MusicStore/Program.cs [hunk 1]
```diff
@@ -17,14 +17,9 @@ public Task<int> Main(string[] args)
...
 
-            var context = new HostingContext()
-            {
-                Configuration = config,
-                ServerFactoryLocation = "Microsoft.AspNet.Server.WebListener",
-                ApplicationName = "MusicStore"
-            };
-
-            using (new HostingEngine().Start(context))
+            using (WebApplication.CreateHostingEngine(config)
```
Comment (Praburaj) [inline src/MusicStore/Program.cs]: Can't we bring back `WebApp.Start<T>`?

Comment (HaoK) [inline src/MusicStore/Program.cs]: T is the startup type?  @lodejard suggested that belongs as a TestServer extension instead, I can bring it up with him again

Comment (Praburaj) [inline src/MusicStore/Program.cs]: >T is the startup type

yeah.

Comment (HaoK) [inline src/MusicStore/Program.cs]: I'm not sure what change you are advocating for music store though,  you don't like how hosting automatically scans for startup today?  You would prefer something more explicit like WebApplication.CreateHostingEngine(config).Start<Startup>()?

Comment (Praburaj) [inline src/MusicStore/Program.cs]: oh no. That question was in general. Not specific to MusicStore.

**Pull request closed without merging**